test/test_build.py

```diff
@@ -52,23 +54,47 @@ def image_type_fixture(shared_tmpdir, build_container, request, force_aws_upload
     Build an image inside the passed build_container and return an
     ImageBuildResult with the resulting image path and user/password
     """
+    with build_images(shared_tmpdir, build_container, request, force_aws_upload) as build_results:
+        yield build_results[0]
+
+
+@pytest.fixture(name="images", scope="session")
+def images_fixture(shared_tmpdir, build_container, request, force_aws_upload):
+    """
+    Build one or more images inside the passed build_container and return an
+    ImageBuildResult array with the resulting image path and user/password
+    """
+    with build_images(shared_tmpdir, build_container, request, force_aws_upload) as build_results:
+        yield build_results
+
+
+@contextmanager
+def build_images(shared_tmpdir, build_container, request, force_aws_upload):
+    """
+    Build all available image types if necessary and return the results for the image types that were requested.
+    Will return cached results of previous build requests.
+    """
     # image_type is passed via special pytest parameter fixture
     if request.param.count(",") == 2:
-        container_ref, image_type, target_arch = request.param.split(",")
+        container_ref, images, target_arch = request.param.split(",")
     elif request.param.count(",") == 1:
-        container_ref, image_type = request.param.split(",")
+        container_ref, images = request.param.split(",")
         target_arch = None
     else:
         raise ValueError(f"cannot parse {request.param.count}")
 
+    # images might be multiple --type args
+    # split and check each one
+    image_types = images.split("+")
+
     username = "test"
     password = "password"
 
-    # image_type can be long and the qmp socket (that has a limit of 100ish
-    # AF_UNIX) is derrived from the path
-    # so hash the image_type instead of just using it
-    for_image_type = request.param
-    output_path = shared_tmpdir / format(abs(hash(for_image_type)), "x")
+    # params can be long and the qmp socket (that has a limit of 100ish
```

Comment:
Given that we build them all now the caching can be simplified I think, I will double check and (maybe) do a followup with that